iceoryx_examples/icedelivery/iox_subscriber_typed_modern.cpp

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iceoryx_posh/popo/modern_api/typed_subscriber.hpp"
+#include "iceoryx_posh/popo/modern_api/untyped_subscriber.hpp"
+#include "iceoryx_posh/popo/guard_condition.hpp"
+#include "iceoryx_posh/popo/wait_set.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+#include "topic_data.hpp"
+
+#include <chrono>
+#include <csignal>
+#include <iostream>
+
+bool killswitch = false;
+iox::popo::GuardCondition shutdownGuard;
+
+static void sigHandler(int f_sig [[gnu::unused]])
+{
+    // caught SIGINT, now exit gracefully
+    killswitch = true;
+    shutdownGuard.trigger(); // unblock waitsets
+}
+
+void subscriberHandler(iox::popo::WaitSet& waitSet)
+{
+    // run until interrupted
+    while(!killswitch)
+    {
+        auto triggeredConditions = waitSet.wait();
+        for(auto& condition : triggeredConditions)
+        {
+            auto subscriber = dynamic_cast<iox::popo::TypedSubscriber<Position>*>(condition);
+            if(subscriber)
+            {
+                subscriber->take().and_then([](iox::cxx::optional<iox::popo::Sample<const Position>>& maybePosition){
```

Comment:
As discussed it has a complicated touch for me 
take -> and_then -> has_value -> value -> now we are ready to read. The question is if we can make it simpler without loosing the strength of expected and optional

Reply:
Will be addressed in #340